components/dashboards-web-component/src/designer/components/WidgetConfigurationPanel.jsx

```diff
@@ -106,15 +202,158 @@ class WidgetConfigurationPanel extends React.Component {
         return publishers;
     }
 
+    getPreferences() {
+        let preferences = [];
+        if (!(dashboardLayout.selectedItem && dashboardLayout.selectedItem.config.content[0].props.configs)) {
+            return <div>No Configurable Options</div>;
+        }
+        let widgetId = dashboardLayout.selectedItem.config.content[0].props.id;
+        let dashboard = this.props.getDashboard();
+        let pageId = this.props.getPageId();
+        let page = DashboardUtils.getPage(dashboard, pageId);
+        let widget = DashboardUtils.searchForWidget(page, widgetId);
+        let options = widget.props.configs.options;
+
+        if (this.state.id !== widgetId) {
+            let newState = this.state;
+            newState.options = options;
+            newState.id = widgetId;
+            this.setState(newState);
+        }
+
+        if (options) {
+            for (let i = 0; i < options.length; i++) {
+                switch (options[i].type) {
+                    case "TypeText":
```

Comment:
AFAIR, in code reviews it was suggested to change enum `TypeText` to `TEXT`.

Apply for others as well; `TypeEnum` --> `ENUM`, `TypeBoolean` --> `BOOLEAN`